### PR TITLE
#92 - Generalize input of validated integ

### LINKED ISSUE #92: Generalize validated ODEs to receive an initial Taylor model
The method https://github.com/JuliaIntervals/TaylorModels.jl/blob/4dd5d17555b375386d58282b64507e1437869fd7/src/validatedODEs.jl#L401

currently receives `(qq0, δq0)` representing a box. It would be nice if it can handle an initial Taylor model reach-set (TaylorModel1 of TaylorN's); the conversion from a box to a TM can be done as a pre-processing step.

### PR DESCRIPTION
**Pull request by mforets**
Closes #92.

### CODE REVIEW

**Comment by mforets** [comment]
i've updated the title of the PR because the branch is still WIP: some additional arguments need to be passed to the initialization functions (eg. `orderT`) while for others such as `symIBox` i think it's better to define it as a function, say `symIbox(N) = ...`.

**Comment by mforets** [comment]
This is now ready for review. Tests pass locally (julia v1.5.2).

File: src/validatedODEs.jl [hunk 1]
```diff
@@ -397,58 +397,122 @@ function validated_step!(f!, t::Taylor1{T}, x::Vector{Taylor1{TaylorN{T}}},
...
+box to the domain `[-1, 1]^n`.
+"""
+function initialize!(X0::IntervalBox{N, T}, orderQ, orderT, x, dx, xTMN, xI, dxI, rem, xTM1v) where {N, T}
+    @assert N == get_numvars()
+
+    # center of the box and vector of widths
+    q0 = mid.(X0)
+    δq0 = X0 .- q0
+
+    # nomalized domain
```
Comment (mforets) [inline src/validatedODEs.jl:413]:
```suggestion
    # normalized domain
```

**Comment by lbenet** [comment]
Thanks a lot for addressing #92!

LGTM. Could you perhaps add some tests for the new method? (Maybe simply adapt the existing ones?)

**Comment by mforets** [comment]
> for others such as the symmetric intreval box I think it's better to define it as a function, say symIbox(N) = ....

this could be defined in IntervalArithmetic.jl (see https://github.com/JuliaIntervals/IntervalArithmetic.jl/pull/446)

**Comment by mforets** [comment]
Should we merge this? Or you'd like to add some further tests?

**Comment by lbenet** [comment]
I would prefer to have some tests... it always helps!

**Comment by mforets** [comment]
The [last commit](https://github.com/JuliaIntervals/TaylorModels.jl/pull/103/commits/ecfb9fc68b0de4363f32fd3f95ad95cf10e06e4a) adds 2 new tests for the functionality of starting `validated_integ` with a vector of TMs.

Do you have some other dynamical system in mind to make a new test?

**Review by lbenet — APPROVED** [review]
LGTM! Thanks a lot!

**Comment by lbenet** [comment]
I'll merge it and release a new patch version. Thanks again!

**Pull request merged**